lib/author.rb

```diff
@@ -0,0 +1,39 @@
+class Author
+  attr_reader :first_name,
+              :last_name,
+              :books
+
+  def initialize(attributes)
+    @first_name = attributes[:first_name]
+    @last_name = attributes[:last_name]
+    @books = []
+  end
+
+  def name
+    "#{@first_name} #{@last_name}"
+  end
+
+  def add_book(book)
+    @books << book
+  end
+
+  def class
```

Comment:
When we call `thing.class` it will tell us the _class_ of object that thing is.  So `[1, 2, 3].class` returns `Array`.  By overriding the class method, you have made your job more difficult!  In the Author's `write` method, an instance of Book should be created, and that object will, inherently, know that it is a Book - we don't need to recreate that knowledge.